runtime/tests/port/j9strTest.c

```diff
@@ -1604,35 +1609,71 @@ j9str_test_atoe_vsnprintf(struct J9PortLibrary *portLibrary)
-	rc |= printToBuffer(portLibrary, FALSE, buffer1, bufferLength1, expectedOutput1, "%*s", 4, nullTerminatedString);
+	rc |= printToBuffer(portLibrary, FALSE, buffer1, bufferLength1, expectedOutput1, strlen(expectedOutput1), "%*s", 4, nullTerminatedString);
 
 	/* min_width is greater than the length of inputString (buffer length > min_width) */
 	j9tty_printf(PORTLIB, "\n\tTesting case 4\n");
-	rc |= printToBuffer(portLibrary, FALSE, buffer3, bufferLength3, expectedOutput3, "%*s", 10, nullTerminatedString);
+	rc |= printToBuffer(portLibrary, FALSE, buffer3, bufferLength3, expectedOutput3, strlen(expectedOutput3), "%*s", 10, nullTerminatedString);
 
 	/* precision is equal to the length of inputString */
 	j9tty_printf(PORTLIB, "\n\tTesting case 5\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, "%.*s", 4, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, strlen(expectedOutput1), "%.*s", 4, nonNullTerminatedString);
 
 	/* precision is less than the length of inputString */
 	j9tty_printf(PORTLIB, "\n\tTesting case 6\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput4, "%.*s", 2, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput4, strlen(expectedOutput4), "%.*s", 2, nonNullTerminatedString);
 
 	/* both min_width and precision are equal to the length of inputString */
 	j9tty_printf(PORTLIB, "\n\tTesting case 7\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, "%*.*s", 4, 4, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput1, strlen(expectedOutput1), "%*.*s", 4, 4, nonNullTerminatedString);
 
 	/* the length of inputString is equal to precision but less than min_width */
 	j9tty_printf(PORTLIB, "\n\tTesting case 8\n");
-	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput3, "%*.*s", 10, 4, nonNullTerminatedString);
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, expectedOutput3, strlen(expectedOutput3), "%*.*s", 10, 4, nonNullTerminatedString);
+
+	/* the buffer is NULL. expect to estimate the length */
+	j9tty_printf(PORTLIB, "\n\tTesting case 9\n");
+	rc |= printToBuffer(portLibrary, FALSE, NULL, 0, NULL, strlen(nonNullTerminatedString), "%s", nonNullTerminatedString);
+
+	/* number with min_width */
+	j9tty_printf(PORTLIB, "\n\tTesting case 10\n");
+	rc |= printToBuffer(portLibrary, FALSE, buffer0, bufferLength0, "  -123", 6, "%*d", 6, -123);
+
+	/* number with precision */
+	j9tty_printf(PORTLIB, "\n\tTesting case 11\n");
+	rc |= printToBuffer(portLibrary, TRUE, buffer0, bufferLength0, "-000123", 7, "%.*d", 6, -123);
+
+	/* long number with min_width */
+	j9tty_printf(PORTLIB, "\n\tTesting case 12\n");
+	rc |= printToBuffer(portLibrary, FALSE, buffer0, bufferLength0, "  -1234", 7, "%*ld", 7, -1234l);
```

Comment:
Please use uppercase `L` instead of `l` which in some fonts is indistinguishable from `1`.

Reply:
Thanks @keithc-ca for your review. I addressed all comments on my latest commit. If it looks good to you, I squash my commits.